Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl

```diff
@@ -229,8 +229,13 @@ void main( uint3 DTid : SV_DispatchThreadID )
 
                 hierarchyBuffer[partition.NodeIndex].LeftChildIndex = leftEntry.NodeIndex;
                 hierarchyBuffer[partition.NodeIndex].RightChildIndex = rightEntry.NodeIndex;
-                hierarchyBuffer[leftEntry.NodeIndex].ParentIndex = partition.NodeIndex;
+                hierarchyBuffer[leftEntry.NodeIndex].ParentIndex = partition.NodeIndex;                
                 hierarchyBuffer[rightEntry.NodeIndex].ParentIndex = partition.NodeIndex;
+                if (Constants.UpdatesAllowed) 
+                {
+                    aabbParentBuffer[leftEntry.NodeIndex] = partition.NodeIndex;
+                    aabbParentBuffer[rightEntry.NodeIndex] = partition.NodeIndex;
```

Comment:
I'm not a big fan of this whole game of making sure this parent buffer is updated every time the hierarchy buffer is being updated, and I'm not convinced any real savings is happening. Any reason we can't populate these buffers after the hierarchy buffer is finalized?